src/main/java/com/johnstarich/moviematcher/models/User.java

```diff
@@ -94,63 +95,80 @@ public User resetPassword(String oldPassword, String newPassword) throws HttpExc
 			throw new HttpException(HttpStatus.BAD_REQUEST, "No previous password set.");
 		}
 		if(! BCrypt.checkpw(oldPassword, password)) {
-			throw new HttpException(HttpStatus.BAD_REQUEST, "Invalid password.");
+			throw new HttpException(HttpStatus.BAD_REQUEST, "Invalid old password.");
 		}
 
-		User u = new User(id, email, first_name, last_name, friends, groups, BCrypt.hashpw(newPassword, BCrypt.gensalt()));
+		User u = new User(id, username, first_name, last_name, friends, groups, BCrypt.hashpw(newPassword, BCrypt.gensalt()));
 		return u.save();
 	}
 
-	public static User loadByUsername(String email) {
-		return MovieMatcherDatabase.morphium.createQueryFor(User.class).f("email").eq(email).get();
+	public static Optional<User> loadByUsername(String username) {
+		List<User> users = MovieMatcherDatabase.morphium.createQueryFor(User.class).f("username").eq(username).limit(1).asList();
+		if(users.size() == 0 || users.get(0) == null) {
+			return Optional.empty();
+		}
+		return Optional.of(users.get(0));
+	}
+
+	public static Session login(String username, String password) throws HttpException {
+		Optional<User> user = loadByUsername(username);
+		// username does not exist
+		if(! user.isPresent()) {
+			throw new HttpException(HttpStatus.UNAUTHORIZED, "Bad username/password combination. Please try again.");
+		}
+		// passwords do not match
+		if(! BCrypt.checkpw(password, user.get().password)) {
+			throw new HttpException(HttpStatus.UNAUTHORIZED, "Bad username/password combination. Please try again.");
+		}
+		return new Session(null, user.get()).save();
```

Comment:
It does the same thing as `new ObjectId()` but it's one less import
I can change it if you think I should, for consistency's sake or something